src/caw/functions/caw_functions.cpp

```diff
@@ -0,0 +1,246 @@
+// This is a file that will contain the caw functions
+// to be registered with, and used by Faz
+#include "caw_functions.h"
+
+#include <chrono>
+#include <ctime>
+#include <sstream>
+#include <string>
+#include <sys/time.h>
+#include <vector>
+
+bool isInList(const std::vector<std::string> &list, const std::string &key) {
+  for (std::string s : list) {
+    if (s == key) {
+      return true;
+    }
+  }
+  return false;
+}
+
+std::vector<std::string> stringToVector(const std::string &strToConvert) {
+  std::stringstream ss(strToConvert);
+  std::vector<std::string> toReturn;
+  std::string word;
+  while (ss >> word) {
+    toReturn.push_back(word);
+  }
+  return toReturn;
+}
+
+void makeCawFromId(Caw *caw, const std::string &caw_id,
+                   KeyValueStoreInterface &kvstore) {
+  std::vector<std::string> currentCaw;
+  Timestamp *timestamp = new Timestamp;
+  std::vector<std::string> timestamps;
+
+  currentCaw = kvstore.Get("caw_" + caw_id);
+
+  caw->set_username(currentCaw[0]);
+  caw->set_text(currentCaw[1]);
+  caw->set_id(caw_id);
+  caw->set_parent_id(currentCaw[2]);
+  // Construct the timestamp and set it
+  timestamps = stringToVector(currentCaw[3]);
+  timestamp->set_seconds(std::stol(timestamps[0]));
+  timestamp->set_useconds(std::stol(timestamps[1]));
+  caw->set_allocated_timestamp(timestamp);
+}
+
+Status RegisterUser(const Any &EventRequest, Any &EventReply,
+                    KeyValueStoreInterface &kvstore) {
+  RegisteruserRequest request;
+  RegisteruserReply response;
+  EventRequest.UnpackTo(&request);
+  std::vector<std::string> userList = kvstore.Get("caw_users");
+  Status status;
+  if (isInList(userList, request.username())) {
+    status = Status(StatusCode::ALREADY_EXISTS,
+                    "The provided username already exists.");
+  } else {
+    kvstore.Put("caw_users", request.username());
+    status = Status::OK;
+  }
+  EventReply.PackFrom(response);
+  return status;
+}
+
+Status PostCaw(const Any &EventRequest, Any &EventReply,
+               KeyValueStoreInterface &kvstore) {
+  CawRequest request;
+  CawReply response;
+  EventRequest.UnpackTo(&request);
+  std::vector<std::string> userList = kvstore.Get("caw_users");
+  std::vector<std::string> all_caws = kvstore.Get("all_caws");
```

Comment:
Have removed the all_caws list. I am not storing usernames as keys (because according to our current specs we never want to store a value associated with username) so I am still keeping the caw_users vector.